agent/consul/state/catalog_oss.go

```diff
@@ -208,14 +208,18 @@ func (s *Store) catalogInsertService(tx *memdb.Txn, svc *structs.ServiceNode) er
 		return fmt.Errorf("failed inserting service: %s", err)
 	}
 
-	// overall services index
-	if err := tx.Insert("index", &IndexEntry{"services", svc.ModifyIndex}); err != nil {
-		return fmt.Errorf("failed updating index: %s", err)
+	if err := s.catalogUpdateServicesIndexes(tx, svc.ModifyIndex, &svc.EnterpriseMeta); err != nil {
```

Comment:
`catalogUpdateServicesIndexes` is something different from `catalogUpdateServiceIndexes` below?

Reply:
Yes the one is to update the `services` index. The other is to update the per-service `service.<service name>`